<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from /home/chris.paxton/catkin_ws/src/costar_plan/costar_simulation/robot/ur5_joint_limited_robot.xacro | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="ur5_robotiq_parallel" xmlns:xacro="http://www.ros.org/wiki/xacro">
  <!-- common stuff -->
  <gazebo>
    <plugin filename="libgazebo_ros_control.so" name="ros_control">
      <!--robotNamespace>/</robotNamespace-->
      <!--robotSimType>gazebo_ros_control/DefaultRobotHWSim</robotSimType-->
    </plugin>
    <!--
    <plugin name="gazebo_ros_power_monitor_controller" filename="libgazebo_ros_power_monitor.so">
      <alwaysOn>true</alwaysOn>
      <updateRate>1.0</updateRate>
      <timeout>5</timeout>
      <powerStateTopic>power_state</powerStateTopic>
      <powerStateRate>10.0</powerStateRate>
      <fullChargeCapacity>87.78</fullChargeCapacity>     
      <dischargeRate>-474</dischargeRate>
      <chargeRate>525</chargeRate>
      <dischargeVoltage>15.52</dischargeVoltage>
      <chargeVoltage>16.41</chargeVoltage>
    </plugin>
-->
  </gazebo>
  <!-- ur5 -->
  <!-- Inertia parameters -->
  <!-- This mass might be incorrect -->
  <!-- Kinematic model -->
  <!-- Properties from urcontrol.conf -->
  <!--
    DH for UR5:
    a = [0.00000, -0.42500, -0.39225,  0.00000,  0.00000,  0.0000]
    d = [0.089159,  0.00000,  0.00000,  0.10915,  0.09465,  0.0823]
    alpha = [ 1.570796327, 0, 0, 1.570796327, -1.570796327, 0 ]
    q_home_offset = [0, -1.570796327, 0, -1.570796327, 0, 0]
    joint_direction = [-1, -1, 1, 1, 1, 1]
    mass = [3.7000, 8.3930, 2.2750, 1.2190, 1.2190, 0.1879]
    center_of_mass = [ [0, -0.02561, 0.00193], [0.2125, 0, 0.11336], [0.11993, 0.0, 0.0265], [0, -0.0018, 0.01634], [0, 0.0018,0.01634], [0, 0, -0.001159] ]
  -->
  <!-- Arbitrary offsets for shoulder/elbow joints -->
  <!-- measured from model -->
  <!-- measured from model -->
  <!-- link lengths used in model -->
  <!--property name="shoulder_height" value="0.089159" /-->
  <!--property name="shoulder_offset" value="0.13585" /-->
  <!-- shoulder_offset - elbow_offset + wrist_1_length = 0.10915 -->
  <!--property name="upper_arm_length" value="0.42500" /-->
  <!--property name="elbow_offset" value="0.1197" /-->
  <!-- CAD measured -->
  <!--property name="forearm_length" value="0.39225" /-->
  <!--property name="wrist_1_length" value="0.093" /-->
  <!-- CAD measured -->
  <!--property name="wrist_2_length" value="0.09465" /-->
  <!-- In CAD this distance is 0.930, but in the spec it is 0.09465 -->
  <!--property name="wrist_3_length" value="0.0823" /-->
  <!-- manually measured -->
  <!-- manually measured -->
  <!-- manually measured -->
  <!-- manually measured -->
  <!-- manually measured -->
  <!-- Robotiq gripper -->
  <link name="base_link">
    <visual>
      <origin rpy="1.570795 0 0" xyz="0.0 0.0 0.0"/>
      <geometry>
        <mesh filename="package://costar_simulation/meshes/ur5/visual/Base.dae.obj"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://costar_simulation/meshes/ur5/collision/base.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="4.0"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.00443333156" ixy="0.0" ixz="0.0" iyy="0.00443333156" iyz="0.0" izz="0.0072"/>
    </inertial>
  </link>
  <joint name="shoulder_pan_joint" type="revolute">
    <parent link="base_link"/>
    <child link="shoulder_link"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.089159"/>
    <axis xyz="0 0 1"/>
    <limit effort="150.0" lower="-3.14159" upper="3.14159" velocity="3.15"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="shoulder_link">
    <visual>
      <origin rpy="1.570795 0 0" xyz="0.0 0.0 0.0"/>
      <geometry>
        <mesh filename="package://costar_simulation/meshes/ur5/visual/Shoulder.dae.obj"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://costar_simulation/meshes/ur5/collision/shoulder.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="3.7"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.010267495893" ixy="0.0" ixz="0.0" iyy="0.010267495893" iyz="0.0" izz="0.00666"/>
    </inertial>
  </link>
  <joint name="shoulder_lift_joint" type="revolute">
    <parent link="shoulder_link"/>
    <child link="upper_arm_link"/>
    <origin rpy="0.0 1.570795 0.0" xyz="0.0 0.13585 0.0"/>
    <axis xyz="0 1 0"/>
    <limit effort="150.0" lower="-3.14159" upper="3.14159" velocity="3.15"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="upper_arm_link">
    <visual>
      <origin rpy="1.570795 0 0" xyz="0.0 0.0 0.0"/>
      <geometry>
        <mesh filename="package://costar_simulation/meshes/ur5/visual/UpperArm.dae.obj"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://costar_simulation/meshes/ur5/collision/upper_arm.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="8.393"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.28"/>
      <inertia ixx="0.22689067591" ixy="0.0" ixz="0.0" iyy="0.22689067591" iyz="0.0" izz="0.0151074"/>
    </inertial>
  </link>
  <joint name="elbow_joint" type="revolute">
    <parent link="upper_arm_link"/>
    <child link="forearm_link"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 -0.1197 0.425"/>
    <axis xyz="0 1 0"/>
    <limit effort="150.0" lower="-3.14159" upper="3.14159" velocity="3.15"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="forearm_link">
    <visual>
      <origin rpy="1.570795 0 0" xyz="0.0 0.0 0.0"/>
      <geometry>
        <mesh filename="package://costar_simulation/meshes/ur5/visual/Forearm.dae.obj"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://costar_simulation/meshes/ur5/collision/forearm.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="2.275"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.25"/>
      <inertia ixx="0.049443313556" ixy="0.0" ixz="0.0" iyy="0.049443313556" iyz="0.0" izz="0.004095"/>
    </inertial>
  </link>
  <joint name="wrist_1_joint" type="revolute">
    <parent link="forearm_link"/>
    <child link="wrist_1_link"/>
    <origin rpy="0.0 1.570795 0.0" xyz="0.0 0.0 0.39225"/>
    <axis xyz="0 1 0"/>
    <limit effort="28.0" lower="-3.14159" upper="3.14159" velocity="3.2"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="wrist_1_link">
    <visual>
      <origin rpy="1.570795 0 0" xyz="0.0 0.0 0.0"/>
      <geometry>
        <mesh filename="package://costar_simulation/meshes/ur5/visual/Wrist1.dae.obj"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://costar_simulation/meshes/ur5/collision/wrist_1.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="1.219"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.111172755531" ixy="0.0" ixz="0.0" iyy="0.111172755531" iyz="0.0" izz="0.21942"/>
    </inertial>
  </link>
  <joint name="wrist_2_joint" type="revolute">
    <parent link="wrist_1_link"/>
    <child link="wrist_2_link"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.093 0.0"/>
    <axis xyz="0 0 1"/>
    <limit effort="28.0" lower="-3.14159" upper="3.14159" velocity="3.2"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="wrist_2_link">
    <visual>
      <origin rpy="1.570795 0 0" xyz="0.0 0.0 0.0"/>
      <geometry>
        <mesh filename="package://costar_simulation/meshes/ur5/visual/Wrist2.dae.obj"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://costar_simulation/meshes/ur5/collision/wrist_2.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="1.219"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.111172755531" ixy="0.0" ixz="0.0" iyy="0.111172755531" iyz="0.0" izz="0.21942"/>
    </inertial>
  </link>
  <joint name="wrist_3_joint" type="revolute">
    <parent link="wrist_2_link"/>
    <child link="wrist_3_link"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.09465"/>
    <axis xyz="0 1 0"/>
    <limit effort="28.0" lower="-3.14159" upper="3.14159" velocity="3.2"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="wrist_3_link">
    <visual>
      <origin rpy="1.570795 0 0" xyz="0.0 0.0 0.0"/>
      <geometry>
        <mesh filename="package://costar_simulation/meshes/ur5/visual/Wrist3.dae.obj"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://costar_simulation/meshes/ur5/collision/wrist_3.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.1879"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.0171364731454" ixy="0.0" ixz="0.0" iyy="0.0171364731454" iyz="0.0" izz="0.033822"/>
    </inertial>
  </link>
  <joint name="ee_fixed_joint" type="fixed">
    <parent link="wrist_3_link"/>
    <child link="ee_link"/>
    <origin rpy="0.0 0.0 1.570795" xyz="0.0 0.0823 0.0"/>
  </joint>
  <link name="ee_link">
    <collision>
      <geometry>
        <box size="0.01 0.01 0.01"/>
      </geometry>
      <origin rpy="0 0 0" xyz="-0.01 0 0"/>
    </collision>
  </link>
  <transmission name="shoulder_pan_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="shoulder_pan_joint">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="shoulder_pan_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="shoulder_lift_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="shoulder_lift_joint">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="shoulder_lift_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="elbow_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="elbow_joint">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="elbow_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="wrist_1_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="wrist_1_joint">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="wrist_1_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="wrist_2_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="wrist_2_joint">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="wrist_2_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="wrist_3_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="wrist_3_joint">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="wrist_3_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <!-- nothing to do here at the moment -->
  <!-- ROS base_link to UR 'Base' Coordinates transform -->
  <link name="base"/>
  <joint name="base_link-base_fixed_joint" type="fixed">
    <!-- NOTE: this rotation is only needed as long as base_link itself is
                 not corrected wrt the real robot (ie: rotated over 180
                 degrees)
      -->
    <origin rpy="0 0 -3.14159" xyz="0 0 0"/>
    <parent link="base_link"/>
    <child link="base"/>
  </joint>
  <!-- Frame coincident with all-zeros TCP on UR controller -->
  <link name="tool0"/>
  <joint name="wrist_3_link-tool0_fixed_joint" type="fixed">
    <origin rpy="-1.570795 0 0" xyz="0 0.0823 0"/>
    <parent link="wrist_3_link"/>
    <child link="tool0"/>
  </joint>
  <joint name="robotiq_85_base_joint" type="fixed">
    <parent link="ee_link"/>
    <child link="robotiq_85_base_link"/>
    <origin rpy="1.570795 0 0" xyz="0.0 0.0 0.0"/>
  </joint>
  <link name="robotiq_85_base_link">
    <visual>
      <geometry>
        <mesh filename="package://costar_simulation/robot/robotiq_85_gripper/meshes/visual/robotiq_85_base_link.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://costar_simulation/robot/robotiq_85_gripper/meshes/collision/robotiq_85_base_link.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.636951"/>
      <origin xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.000380" ixy="0.000000" ixz="0.000000" iyx="0.000000" iyy="0.001110" iyz="0.000000" izx="0.000000" izy="0.000000" izz="0.001171"/>
    </inertial>
  </link>
  <joint name="robotiq_85_left_knuckle_joint" type="revolute">
    <parent link="robotiq_85_base_link"/>
    <child link="robotiq_85_left_knuckle_link"/>
    <axis xyz="0 0 1"/>
    <origin rpy="3.14159 0.0 0.0" xyz="0.05490451627 0.03060114443 0.0"/>
    <limit effort="1000" lower="0.0" upper="0.804" velocity="2.0"/>
  </joint>
  <joint name="robotiq_85_right_knuckle_joint" type="continuous">
    <parent link="robotiq_85_base_link"/>
    <child link="robotiq_85_right_knuckle_link"/>
    <axis xyz="0 0 1"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.05490451627 -0.03060114443 0.0"/>
    <limit effort="1000" lower="-3.14" upper="3.14" velocity="100.0"/>
    <mimic joint="robotiq_85_left_knuckle_joint" multiplier="1"/>
  </joint>
  <link name="robotiq_85_left_knuckle_link">
    <visual>
      <geometry>
        <mesh filename="package://costar_simulation/robot/robotiq_85_gripper/meshes/visual/robotiq_85_knuckle_link.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://costar_simulation/robot/robotiq_85_gripper/meshes/collision/robotiq_85_knuckle_link.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.018491"/>
      <origin xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.000009" ixy="-0.000001" ixz="0.000000" iyx="-0.000001" iyy="0.000001" iyz="0.000000" izx="0.000000" izy="0.000000" izz="0.000010"/>
    </inertial>
  </link>
  <link name="robotiq_85_right_knuckle_link">
    <visual>
      <geometry>
        <mesh filename="package://costar_simulation/robot/robotiq_85_gripper/meshes/visual/robotiq_85_knuckle_link.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://costar_simulation/robot/robotiq_85_gripper/meshes/collision/robotiq_85_knuckle_link.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.018491"/>
      <origin xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.000009" ixy="-0.000001" ixz="0.000000" iyx="-0.000001" iyy="0.000001" iyz="0.000000" izx="0.000000" izy="0.000000" izz="0.000010"/>
    </inertial>
  </link>
  <joint name="robotiq_85_left_finger_joint" type="fixed">
    <parent link="robotiq_85_left_knuckle_link"/>
    <child link="robotiq_85_left_finger_link"/>
    <origin rpy="0 0 0" xyz="-0.00408552455 -0.03148604435 0.0"/>
  </joint>
  <joint name="robotiq_85_right_finger_joint" type="fixed">
    <parent link="robotiq_85_right_knuckle_link"/>
    <child link="robotiq_85_right_finger_link"/>
    <origin rpy="0 0 0" xyz="-0.00408552455 -0.03148604435 0.0"/>
  </joint>
  <link name="robotiq_85_left_finger_link">
    <visual>
      <geometry>
        <mesh filename="package://costar_simulation/robot/robotiq_85_gripper/meshes/visual/robotiq_85_finger_link.dae"/>
      </geometry>
    </visual>
    <!-- TODO(cpaxton): re-enable this
            <collision>
                <geometry>
                    <mesh filename="package://costar_simulation/robot/robotiq_85_gripper/meshes/collision/robotiq_85_finger_link.stl"/>
                </geometry>
            </collision>
            -->
    <inertial>
      <mass value="0.027309"/>
      <origin xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.000003" ixy="-0.000002" ixz="0.000000" iyx="-0.000002" iyy="0.000021" iyz="0.000000" izx="0.000000" izy="0.000000" izz="0.000020"/>
    </inertial>
  </link>
  <link name="robotiq_85_right_finger_link">
    <visual>
      <geometry>
        <mesh filename="package://costar_simulation/robot/robotiq_85_gripper/meshes/visual/robotiq_85_finger_link.dae"/>
      </geometry>
    </visual>
    <!-- TODO(cpaxton): re-enable this
            <collision>
                <geometry>
                    <mesh filename="package://costar_simulation/robot/robotiq_85_gripper/meshes/collision/robotiq_85_finger_link.stl"/>
                </geometry>
            </collision>
            -->
    <inertial>
      <mass value="0.027309"/>
      <origin xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.000003" ixy="-0.000002" ixz="0.000000" iyx="-0.000002" iyy="0.000021" iyz="0.000000" izx="0.000000" izy="0.000000" izz="0.000020"/>
    </inertial>
  </link>
  <joint name="robotiq_85_left_inner_knuckle_joint" type="continuous">
    <parent link="robotiq_85_base_link"/>
    <child link="robotiq_85_left_inner_knuckle_link"/>
    <axis xyz="0 0 1"/>
    <origin rpy="3.14159 0.0 0.0" xyz="0.06142 0.0127 0"/>
    <limit effort="0.1" lower="-3.14" upper="3.14" velocity="100.0"/>
    <mimic joint="robotiq_85_left_knuckle_joint" offset="0"/>
  </joint>
  <joint name="robotiq_85_right_inner_knuckle_joint" type="continuous">
    <parent link="robotiq_85_base_link"/>
    <child link="robotiq_85_right_inner_knuckle_link"/>
    <axis xyz="0 0 1"/>
    <origin rpy="0 0 0" xyz="0.06142 -0.0127 0"/>
    <limit effort="0.1" lower="-3.14" upper="3.14" velocity="100.0"/>
    <mimic joint="robotiq_85_left_knuckle_joint" offset="0"/>
  </joint>
  <link name="robotiq_85_left_inner_knuckle_link">
    <visual>
      <geometry>
        <mesh filename="package://costar_simulation/robot/robotiq_85_gripper/meshes/visual/robotiq_85_inner_knuckle_link.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://costar_simulation/robot/robotiq_85_gripper/meshes/collision/robotiq_85_inner_knuckle_link.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.029951"/>
      <origin xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.000039" ixy="0.000000" ixz="0.000000" iyx="0.000000" iyy="0.000005" iyz="0.000000" izx="0.000000" izy="0.000000" izz="0.000035"/>
    </inertial>
  </link>
  <link name="robotiq_85_right_inner_knuckle_link">
    <visual>
      <geometry>
        <mesh filename="package://costar_simulation/robot/robotiq_85_gripper/meshes/visual/robotiq_85_inner_knuckle_link.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://costar_simulation/robot/robotiq_85_gripper/meshes/collision/robotiq_85_inner_knuckle_link.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.029951"/>
      <origin xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.000039" ixy="0.000000" ixz="0.000000" iyx="0.000000" iyy="0.000005" iyz="0.000000" izx="0.000000" izy="0.000000" izz="0.000035"/>
    </inertial>
  </link>
  <joint name="robotiq_85_left_finger_tip_joint" type="continuous">
    <parent link="robotiq_85_left_inner_knuckle_link"/>
    <child link="robotiq_85_left_finger_tip_link"/>
    <axis xyz="0 0 1"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.04303959807 -0.03759940821 0.0"/>
    <limit effort="0.1" lower="-3.14" upper="3.14" velocity="100.0"/>
    <mimic joint="robotiq_85_left_knuckle_joint" multiplier="-1"/>
  </joint>
  <joint name="robotiq_85_right_finger_tip_joint" type="continuous">
    <parent link="robotiq_85_right_inner_knuckle_link"/>
    <child link="robotiq_85_right_finger_tip_link"/>
    <axis xyz="0 0 1"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.04303959807 -0.03759940821  0.0"/>
    <limit effort="0.1" lower="-3.14" upper="3.14" velocity="100.0"/>
    <mimic joint="robotiq_85_left_knuckle_joint" multiplier="-1"/>
  </joint>
  <link name="robotiq_85_left_finger_tip_link">
    <visual>
      <geometry>
        <mesh filename="package://costar_simulation/robot/robotiq_85_gripper/meshes/visual/robotiq_85_finger_tip_link.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://costar_simulation/robot/robotiq_85_gripper/meshes/collision/robotiq_85_finger_tip_link.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.019555"/>
      <origin xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.000002" ixy="0.000000" ixz="0.000000" iyx="0.000000" iyy="0.000005" iyz="0.000000" izx="0.000000" izy="0.000000" izz="0.000006"/>
    </inertial>
  </link>
  <link name="robotiq_85_right_finger_tip_link">
    <visual>
      <geometry>
        <mesh filename="package://costar_simulation/robot/robotiq_85_gripper/meshes/visual/robotiq_85_finger_tip_link.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://costar_simulation/robot/robotiq_85_gripper/meshes/collision/robotiq_85_finger_tip_link.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.019555"/>
      <origin xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.000002" ixy="0.000000" ixz="0.000000" iyx="0.000000" iyy="0.000005" iyz="0.000000" izx="0.000000" izy="0.000000" izz="0.000006"/>
    </inertial>
  </link>
  <transmission name="robotiq_85_left_knuckle_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="robotiq_85_left_knuckle_joint">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="robotiq_85_left_knuckle_motor">
      <mechanicalReduction>1</mechanicalReduction>
      <hardwareInterface>PositionJointInterface</hardwareInterface>
    </actuator>
  </transmission>
  <gazebo>
    <!-- loading plugin: mimic joints works in gazebo now -->
    <plugin filename="libroboticsgroup_gazebo_mimic_joint_plugin.so" name="mimic_robotiq_85_1">
      <joint>robotiq_85_left_knuckle_joint</joint>
      <mimicJoint>robotiq_85_right_knuckle_joint</mimicJoint>
      <!--multiplier>1.0</multiplier>
                <offset>0.0</offset-->
    </plugin>
    <plugin filename="libroboticsgroup_gazebo_mimic_joint_plugin.so" name="mimic_robotiq_85_2">
      <joint>robotiq_85_left_knuckle_joint</joint>
      <mimicJoint>robotiq_85_left_inner_knuckle_joint</mimicJoint>
      <!--multiplier>1.0</multiplier>
                <offset>0.0</offset-->
    </plugin>
    <plugin filename="libroboticsgroup_gazebo_mimic_joint_plugin.so" name="mimic_robotiq_85_3">
      <joint>robotiq_85_left_knuckle_joint</joint>
      <mimicJoint>robotiq_85_right_inner_knuckle_joint</mimicJoint>
      <!--multiplier>1.0</multiplier>
                <offset>0.0</offset-->
    </plugin>
    <plugin filename="libroboticsgroup_gazebo_mimic_joint_plugin.so" name="mimic_robotiq_85_4">
      <joint>robotiq_85_left_knuckle_joint</joint>
      <mimicJoint>robotiq_85_left_finger_tip_joint</mimicJoint>
      <multiplier>-1.0</multiplier>
      <offset>0.0</offset>
    </plugin>
    <plugin filename="libroboticsgroup_gazebo_mimic_joint_plugin.so" name="mimic_robotiq_85_5">
      <joint>robotiq_85_left_knuckle_joint</joint>
      <mimicJoint>robotiq_85_right_finger_tip_joint</mimicJoint>
      <multiplier>-1.0</multiplier>
      <offset>0.0</offset>
    </plugin>
  </gazebo>
</robot>

